### PR TITLE
kola/docker: make selinux optional

### PR DESCRIPTION
**Pull request by tormath1**
`selinux` support is going to be disabled on the next shipped Docker.

To keep compatibility with current releases, selinux is made optional:
we just assert that at least `seccomp` or `selinux` or both are
enabled.

this commit will be reverted once SELinux will be back on Flatcar

Signed-off-by: Mathieu Tortuyaux <mathieu@kinvolk.io>

### CODE REVIEW

File: kola/tests/docker/docker.go [hunk 1]
```diff
@@ -610,3 +610,21 @@ func testDockerInfo(expectedFs string, c cluster.TestCluster) {
 		c.Errorf("runc was not in runtimes: %+v", info.Runtimes)
 	}
 }
+
+// hasSecurityOptions strictly checks that at least one of
+// the docker security option is enabled (seccomp, selinux)
+func hasSecurityOptions(opts []string) bool {
+	var selinux, seccomp bool
```
Comment (invidian) [inline kola/tests/docker/docker.go:617]: Wouldn't just `len(opts) > 0` work?

Comment (tormath1) [reply kola/tests/docker/docker.go:617]: Good point ! Previous test was strictly checking for `seccomp` and `selinux` in the security options.

So
```go
len(opts) > 0
```

would work but we won't strictly test the enabled security options. (could be `apparmor`, `toto` or whatever). I tried to stay closer from the current test and to just allow some releases to only have `seccomp` as security options.

File: kola/tests/docker/docker.go [hunk 2]
```diff
@@ -610,3 +610,21 @@ func testDockerInfo(expectedFs string, c cluster.TestCluster) {
 		c.Errorf("runc was not in runtimes: %+v", info.Runtimes)
 	}
 }
+
+// hasSecurityOptions strictly checks that at least one of
+// the docker security option is enabled (seccomp, selinux)
```
Comment (invidian) [inline kola/tests/docker/docker.go:615]:
```suggestion
// the Docker security option is enabled (seccomp, selinux).
```

File: kola/tests/docker/docker.go [hunk 3]
```diff
@@ -610,3 +610,21 @@ func testDockerInfo(expectedFs string, c cluster.TestCluster) {
...
+		case "selinux":
+			selinux = true
+		case "seccomp":
+			seccomp = true
+		default:
+			return false
+		}
+	}
+
+	return selinux || seccomp
```
Comment (invidian) [inline kola/tests/docker/docker.go:629]: Maybe you can simplify to:
```suggestion
	for _, opt := range opts {
		switch opt {
		case "selinux", "seccomp":
			return true
		}
	}

	return false
```

Comment (tormath1) [reply kola/tests/docker/docker.go:629]: Good idea but I'm not sure we won't strictly check for `selinux`, `seccomp` or both. Let's suppose `opts = ["apparmor", "selinux"]` we will return `true` even if we have `apparmor` in the options. :thinking:

Comment (invidian) [reply kola/tests/docker/docker.go:629]: so maybe:
```go
switch opt {
case "selinux", "seccomp":
default:
  return false
}

return true
```

**Comment by tormath1** [comment]
@invidian thanks for your review ! Comments have been addressed :)

**Review by invidian — APPROVED** [review]
Thanks for addressing the feedback @tormath1. Looks good from Go side of things :+1:

**Review by sayanchowdhury — APPROVED** [review]

**Pull request merged**